pkg/roachprod/vm/gce/gcloud.go

```diff
@@ -1649,6 +1649,52 @@ func computeGrowDistribution(groups []jsonManagedInstanceGroup, newNodeCount int
 	return addCount
 }
 
+// Shrink shrinks the cluster by deleting the given VMs. This is only supported
+// for managed instance groups. Currently, nodes should only be deleted from the
+// tail of the cluster, because this function does not yet support renaming
+// nodes and re-syncing DNS names to keep the cluster in a consistent state.
```

Comment:
What would "renaming a node" entail? I'm not sure how we would ensure that roachprod node IDs match crdb's.

Reply:
I updated this comment. But there could be multiple complexities around not deleting from the tail. Roachprod node IDs not matching CRDB's node IDs is not a big deal and more of a nuisance - this happens all the time on DRT, and essentially makes finding the corresponding roachprod node a little bit more tricky (ip lookup). The real issue here would be more towards renaming the nodes so that roachprod is happy with the standard consecutive names it expects as well as updating DNS records both A and SRV. I think we'll ideally want to avoid all this unless there is a good reason to delete start/middle nodes.